### PR TITLE
OHRM5X-1435: Develop recruitment - candidate history API

### PR DESCRIPTION
**Pull request by SandeepaOHRM**

### CODE REVIEW

File: src/plugins/orangehrmRecruitmentPlugin/Dto/CandidateHistorySearchFilterParams.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ */
+
+namespace OrangeHRM\Recruitment\Dto;
+
+use OrangeHRM\Core\Dto\FilterParams;
+use OrangeHRM\ORM\ListSorter;
+
+class CandidateHistorySearchFilterParams extends FilterParams
+{
+    /**
```
Comment (RajithaKumara) [inline src/plugins/orangehrmRecruitmentPlugin/Dto/CandidateHistorySearchFilterParams.php:27]: define allowed sort fields

File: src/plugins/orangehrmRecruitmentPlugin/config/routes.yaml [hunk 2]
```diff
@@ -159,6 +167,21 @@ apiv2_recrutiment_candidates_hire:
...
+  defaults:
+    _api: OrangeHRM\Recruitment\Api\CandidateHistoryAPI
+
+apiv2_recrutiment_candidate_history:
+  path: /api/v2/recruitment/candidates/{candidateId}/history/{historyId}
+  controller: OrangeHRM\Core\Controller\Rest\V2\GenericRestController::handle
+  methods: [ GET, PUT ]
+  defaults:
+    _api: OrangeHRM\Recruitment\Api\CandidateHistoryAPI
+    _key: candidateId
```
Comment (RajithaKumara) [inline src/plugins/orangehrmRecruitmentPlugin/config/routes.yaml:183]:
```suggestion
    _key: historyId
```
:thinking:

File: src/plugins/orangehrmRecruitmentPlugin/entity/Decorator/CandidateHistoryDecorator.php [hunk 3]
```diff
@@ -86,6 +88,13 @@ public function setPerformedBy(int $id): void
...
         $actionId = $this->candidateHistory->getAction();
-        return ucwords(strtolower(CandidateService::STATUS_MAP[$actionId]));
+        $candidateHistoryMap = array_replace(CandidateService::STATUS_MAP, CandidateService::OTHER_ACTIONS_MAP);
+        return ucwords(strtolower($candidateHistoryMap[$actionId]));
+    }
+
+    public function getPerformedDate(): string
+    {
+        $date = $this->candidateHistory->getPerformedDate();
+        return $this->getDateTimeHelper()->formatDateTimeToYmd($date);
```
Comment (RajithaKumara) [inline src/plugins/orangehrmRecruitmentPlugin/entity/Decorator/CandidateHistoryDecorator.php:98]:
```suggestion
        return $this->getDateTimeHelper()->formatDate($date);
```
check this method and continue it

**Review by RajithaKumara — APPROVED** [review]

File: src/plugins/orangehrmRecruitmentPlugin/Dto/CandidateHistorySearchFilterParams.php (hunk 1 above)
Comment (SandeepaOHRM) [reply src/plugins/orangehrmRecruitmentPlugin/Dto/CandidateHistorySearchFilterParams.php:27]: done

File: src/plugins/orangehrmRecruitmentPlugin/config/routes.yaml (hunk 2 above)
Comment (SandeepaOHRM) [reply src/plugins/orangehrmRecruitmentPlugin/config/routes.yaml:183]: done

**Pull request merged**